src-ext/TypeScript/ggvega/src/itmLayerArrayByScales.ts

```diff
@@ -0,0 +1,85 @@
+import * as GG from '../../ggschema/src/index';
+
+import {ItmLayer} from './itmLayer';
+import {VLEncodingField} from './itmEncodingObject';
+import {hasKey, contains} from './utils';
+
+/**
+ * Modify a layer array according to a scale array
+ *
+ * I think the match-and-delete strategy for scales can be very similar
+ * to the one used in {@link itmLayerArrayByLabelsObject}.
+ *
+ * The obvious difference is that scales are in array, rather than in an object.
+ * Another difference is that a scale has an element called `aesthetics` that
+ * can make it simpler to match to the encoding key. Perhaps {@link contains}
+ * could be used.
+ *
+ * **Called by**
+ * @see layerArray
+ *
+ * @param itmLayerArray
+ * @param ggScaleArray
+ *
+ * @returns `ItmLayer[]`
+ */
+export function itmLayerArrayByScalesArray(itmLayerArray: ItmLayer[], ggScaleArray: GG.Scale[]): ItmLayer[] {
+  const scaleMap = {
+    ScaleContinuousPosition: scaleContinuousPosition
+  };
+
+  itmLayerArray.map((itmLayer: ItmLayer) => {
+    for (const encodingKey in itmLayer.encoding) {
+      if (hasKey(itmLayer.encoding, encodingKey)) {
+        ggScaleArray.map((ggScale: GG.Scale) => {
+          //NOTE @wenyu:https://love2dev.com/blog/javascript-remove-from-array/
+
+          for (let i = 0; i < ggScale.aesthetics.length; i++) {
+            if (keyMatch(ggScale.aesthetics[i], encodingKey)) {
+              itmLayer.encoding[encodingKey].title = ggScale.name;
```

Comment:
What if the scale has no name?

Reply:
Oh, That's my fault. 

If `ggScale` doesn't have a name, `ggScale.name = undefined`. Then `encoding.title = undefined`. Then `encoding` will not have a title.  I will fix it.

Reply:
done